src/__fixtures__/json.ts

```diff
@@ -671,6 +671,325 @@ export const JSON_RPC_RESPONSE_FIXTURES = {
   ],
 };
 
+export const JSON_RPC_PENDING_RESPONSE_FIXTURES = {
+  valid: [
+    ...JSON_RPC_SUCCESS_FIXTURES.valid,
+    ...JSON_RPC_FAILURE_FIXTURES.valid,
+    {
+      id: 1,
+      jsonrpc: '2.0',
+    },
+    {
+      id: 1,
+      jsonrpc: '2.0',
+      error: undefined,
+    },
+    {
+      id: 1,
+      jsonrpc: '2.0',
+      result: undefined,
+    },
+    {
+      id: 1,
+      jsonrpc: '2.0',
+      result: undefined,
+      error: undefined,
+    },
+    {
+      id: 1,
+      jsonrpc: '2.0',
+      result: {
+        foo: 'bar',
+      },
+      error: {
+        code: -32000,
+        message: 'Internal error',
+      },
```

Comment:
It appears that this is intended behavior. PendingJsonRpcResponse was created to represent a response object within middleware. The idea here is that while a request is being sent through layers of a middleware stack, we don't know what the final response object will be — we don't know whether it will end up representing a successful or failed response. So PendingJsonRpcResponse is like the Schrodinger's cat of responses. It looks like the possibility that *both* `result` and `error` could be set was acknowledged but accepted as a compromise to prevent from having to typecast the response in middleware: https://github.com/MetaMask/json-rpc-engine/pull/75